app/Policies/AchievementSetClaimPolicy.php

```diff
@@ -31,22 +34,93 @@ public function view(?User $user, AchievementSetClaim $achievementSetClaim): boo
         return true;
     }
 
-    public function create(User $user): bool
+    public function create(User $user, ?Game $game = null): bool
     {
-        return $user->hasAnyRole([
-            Role::DEVELOPER,
-            Role::DEVELOPER_JUNIOR,
-        ]);
+        // First, check the basic role requirement.
+        if (!$user->hasAnyRole([Role::DEVELOPER, Role::DEVELOPER_JUNIOR])) {
+            return false;
+        }
+
+        // If no game is provided, we'll just be happy the user passes role verification.
+        if (!$game) {
+            return true;
+        }
+
```

Comment:
Would it make sense to disallow claiming games for unsupported systems?
http://localhost:64000/game2/7775

```
        $game->loadMissing('system');
        if (!$game->system->active) {
            return false;
        }
```

Reply:
To an extent, I think it would. The only thing I'm not sure about is how to handle systems that are currently undergoing a rollout. Aren't they still considered unsupported? It almost seems like we may need some kind of `are_achievement_sets_claimable` flag on the system model.